journal/src/test/java/io/camunda/zeebe/journal/file/SegmentedJournalTest.java

```diff
@@ -892,6 +897,49 @@ void shouldPreventWriteExclusiveOperationsWhileFlushing() {
     assertThat(journalFactory.metaStore().loadLastFlushedIndex()).isEqualTo(lastWrittenIndex);
   }
 
+  @Test
+  void shouldFailWithOODForAsyncSegmentCreation() {
+    // given
+    journalFactory = new TestJournalFactory("test", 1, createFailingSegmentAllocator(3));
+    journal = journalFactory.journal(journalFactory.segmentsManager(directory));
+    journal.append(1, journalFactory.entry()); // first segment
+    journal.append(2, journalFactory.entry()); // second segment - after this we are creating async
+
+    // when - then
+    assertThatThrownBy(() -> journal.append(3, journalFactory.entry()))
+        .isInstanceOf(OutOfDiskSpace.class)
+        .hasMessage("Nope, no free space.");
+    assertThatThrownBy(() -> journal.append(4, journalFactory.entry()))
+        .isInstanceOf(OutOfDiskSpace.class)
+        .hasMessage("Nope, no free space.");
+  }
+
+  @RegressionTest("https://github.com/camunda/zeebe/issues/13955")
+  void shouldPrintAsyncOODOnlyOnceFailed() {
```

Comment:
:thought_balloon: I'm not sure if this test is all that necessary. What do we want to test? That we switch to synchronous creation, and as such that message is printed out after? I'd like testing that better than log messages =/ That said, I don't have a good idea OTOH, hence the :thought_balloon:

Reply:
Thats the thing I was also struggling. There is not really a behavior observable from outside, which we can verify except the logging.